cdap-explore/src/main/java/co/cask/cdap/explore/guice/ExploreRuntimeModule.java

```diff
@@ -186,13 +186,14 @@ private static File createLocalDir(CConfiguration cConf, String dirName) {
       @Override
       public ExploreService get() {
         File hiveDataDir = new File(cConf.get(Constants.Explore.LOCAL_DATA_DIR));
+        File defaultScratchDir = new File(hiveDataDir, cConf.get(Constants.AppFabric.TEMP_DIR));
 
-        // The properties set using setProperty will be included to any new HiveConf object created,
-        // at the condition that the configuration is known by Hive, and so is one of the HiveConf.ConfVars
-        // variables.
-
-        System.setProperty(HiveConf.ConfVars.SCRATCHDIR.toString(),
-                           new File(hiveDataDir, cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsolutePath());
+        if (System.getProperty(HiveConf.ConfVars.SCRATCHDIR.toString()) == null) {
+          // The properties set using setProperty will be included to any new HiveConf object created,
```

Comment:
The comment applies to all the `System.setProperty()` calls made in this method. It would be better to keep this outside the if condition. Notice that there was a newline between the comment and the first `System.setProperty()` earlier.